### ISSUE TITLE
[Desktop] Mouse capture is lost when dragging an element outside of the app frame

### ISSUE THREAD

**Issue by kucint** [unoplatform/uno#18290]
### Current behavior

When dragging an `UIElement` outside of the application border, the mouse capture is lost. this causes some issues later on.

https://github.com/user-attachments/assets/df34bb04-8012-4ce1-92d3-a1e39316ac38

### Expected behavior

When dragging an `UIElement` outside of the application border and back, the mouse capture shall persist.

### How to reproduce it (as minimally and precisely as possible)

MINIMAL REPRO PROJECT 
[DragAndDropUnoApp.zip](https://github.com/user-attachments/files/17127174/DragAndDropUnoApp.zip)

STEPS TO REPRODUCE
- start the app
- click an item and start to drag. The item is following the mouse movement (is dragged)
- continue to drag the item by moving the mouse outside of the app frame, the item stops to move
- drag back to the app frame. The item is no longer following the mouse movement (**BUG**)
- release the left mouse button and move the mouse. The item is again following the mouse even if mouse button is no loner pressed (**BUG**)
- click on the canvas. series of exceptions are thrown: (**BUG**)
  - _Microsoft.UI.Input.GestureRecognizer: Error: Microsoft.UI.Xaml.Controls.ScrollContentPresenter Inconsistent state, we already have a pending gesture for a pointer that is going down. Abort the previous gesture._
  - _Microsoft.UI.Input.GestureRecognizer: Error: Microsoft.UI.Xaml.Shapes.Rectangle Inconsistent state, we already have a pending gesture for a pointer that is going down. Abort the previous gesture._
  - _Microsoft.UI.Input.GestureRecognizer+Manipulation: Error: Invalid manipulation state: We are receiving a down for the second time for the same pointer!This is however common when using iOS emulator with VNC where we might miss some pointer events due to focus being stole by debugger, in that case you can safely ignore this message._

Please note that this issue occurs only on `Desktop`. On `WindowsSdk` all is OK.

### Workaround

none

### Works on UWP/WinUI

None

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

`"Uno.Sdk": "5.3.108"`

### Affected platforms

Skia (WPF)

### IDE

Visual Studio 2022

**Comment by ramezgerges** [comment]
Dragging an element outside the app calls WPF's `System.Windows.DragDrop.DoDragDrop`, which blocks until the dragging operation is done (dropped or cancelled). So, not only is the pointer capture lost (which is done by Wpf itself), but the entire app will appear "frozen" until dragging is done. I don't think there's a whole lot to do about this (at least on Wpf).

@dr1rrb any thoughts?

**Comment by qian-o** [comment]
I also encountered this problem. After multiple drag-and-drops, an exception will be thrown from Microsoft.UI.Input.GestureRecognizer.

https://github.com/user-attachments/assets/d1a03eed-68d9-4f21-883f-d11655fbb537

**Comment by MartinZikmund** [comment]
@ramezgerges can you please re-test this with the upcoming new "backend"?

**Comment by qian-o** [comment]
> [@ramezgerges](https://github.com/ramezgerges) can you please re-test this with the upcoming new "backend"?

Is Uno going to launch a new backend in the near future? Where can I find information about this?

**Comment by jeromelaban** [comment]
@qian-o the new win32 backend has been released with Uno 6, it might help with this issue.

**Comment by qian-o** [comment]
> [@qian-o](https://github.com/qian-o) the new win32 backend has been released with Uno 6, it might help with this issue.

Yes, I have noticed the new Win32 backend, and it has already addressed previous issues. However, I would like to know how to handle drag-and-drop operations between multiple windows, as this still cannot be managed in Win32 at the moment.

Additionally, after dragging a window out, the drag animation does not follow the cursor. Although this is not a major issue, the animation stops updating once the mouse returns to the window.

**Comment by dr1rrb** [comment]
Drag and drop between applications should be supported on win32 (thanks the [`Win32DragDropExtension`](https://github.com/unoplatform/uno/blob/master/src/Uno.UI.Runtime.Skia.Win32/Win32DragDropExtension.cs)). To use it you have to use the drag and drop API of WinUI, you can look at [Microsoft's documentation](https://learn.microsoft.com/en-us/windows/apps/design/input/drag-and-drop) (or also our [samples in our internal test app](https://github.com/unoplatform/uno/tree/5fe61af91096a73de2b492694218a61052a71169/src/SamplesApp/UITests.Shared/Windows_UI_Xaml/DragAndDrop)).

Also you are mentioning an issue with the "the drag animation does not follow the cursor", could you provided more details?

**Comment by ramezgerges** [comment]
Actually, external drag and drop support on Win32 is not supported. The infrastructure is there but the COM protocol for this is not implemented correctly on our side.

**Comment by qian-o** [comment]
> Drag and drop between applications should be supported on win32 (thanks the [`Win32DragDropExtension`](https://github.com/unoplatform/uno/blob/master/src/Uno.UI.Runtime.Skia.Win32/Win32DragDropExtension.cs)). To use it you have to use the drag and drop API of WinUI, you can look at [Microsoft's documentation](https://learn.microsoft.com/en-us/windows/apps/design/input/drag-and-drop) (or also our [samples in our internal test app](https://github.com/unoplatform/uno/tree/5fe61af91096a73de2b492694218a61052a71169/src/SamplesApp/UITests.Shared/Windows_UI_Xaml/DragAndDrop)).
> 
> Also you are mentioning an issue with the "the drag animation does not follow the cursor", could you provided more details?

I'm sorry for the late reply. I've recorded a video to reproduce the two issues I mentioned.

https://github.com/user-attachments/assets/082d2175-575d-4268-9743-9820ff853bcf

1.	In the first half of the video, it shows the preview effect that follows the mouse during dragging. Currently, it only works within a single window. If you move the mouse outside the window and then back in, the preview effect stops updating.
2.	In the second half of the video, it demonstrates the issue with dragging between multiple windows. Currently, when I drag an element from window A to window B, window B doesn't respond to any drag-and-drop events.

[UnoDragDrop.zip](https://github.com/user-attachments/files/21173837/UnoDragDrop.zip)
